pkg/cache/application.go

```diff
@@ -456,6 +456,29 @@ func (app *Application) handleRecoverApplicationEvent(event *fsm.Event) {
 	}
 }
 
+func (app *Application) skipReservationStage() bool {
+	// no task groups defined, skip reservation
+	if len(app.taskGroups) == 0 {
+		log.Logger().Debug("Skip reservation stage: no task groups defined")
```

Comment:
Add the appID to the log to allow tracking the flow correctly. There might be multiple apps doing the same thing.

Reply:
Make sense, fixed.